### PR TITLE
Add doc for TF Serving

### PR DESCRIPTION
**Pull request by lluunn**
/cc @jlewi 
/cc @ynqa

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1165)

### CODE REVIEW

**Comment by lluunn** [comment]
Tests timeout

**Comment by lluunn** [comment]
#1184

**Comment by lluunn** [comment]
Thank you, PTAL @jlewi

File: docs_dev/tf_serving.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Also, there will be liveness and healthness check. See
+[issue](https://github.com/kubeflow/kubeflow/issues/368).
+
+## Usage
+To use the TF Serving component, see document [here](https://github.com/kubeflow/kubeflow/blob/master/components/k8s-model-server/README.md).
+
+## SeldonIO
+[SeldonIO](https://github.com/SeldonIO/seldon-core) is a platform for serving ML
+models. It's an open question on how should we integrate TF Serving with
```
Comment (jlewi) [inline docs_dev/tf_serving.md]: ... on how should we -> Its an open question how we should

File: docs_dev/tf_serving.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Seldon to get nice features like routing and model management.
+
+## Next steps
+- [kubeflow/kubeflow#1000](https://github.com/kubeflow/kubeflow/issues/1000): Support logging request and response
```
Comment (jlewi) [inline docs_dev/tf_serving.md]: How about autoscaling?

File: docs_dev/tf_serving.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### HTTP proxy
+The [http proxy](https://github.com/kubeflow/kubeflow/tree/master/components/k8s-model-server/http-proxy)
+provides an REST api, which is the same as TF Serving's built-in REST api.
+
+We initially implemented this because TF Serving only supported gRPC api (until
+TF 1.8). Going forward, we will still maintain and support this http proxy as
+it has the ability to provide additional features such as:
+- exporting metrics, [kubeflow/kubeflow#1036](https://github.com/kubeflow/kubeflow/issues/1036).
+- request logging. See
```
Comment (jlewi) [inline docs_dev/tf_serving.md]: Are we planning on doing request logging in the http proxy? Why can't we do that in TF Serving?
Would we have to also proxy gRPC requests in order to get that functionality when gRPC is used?

File: docs_dev/tf_serving.md (hunk 2 above)
Comment (lluunn) [reply docs_dev/tf_serving.md]: There is a benchmark issue below, which should cover it? I added some word

File: docs_dev/tf_serving.md (hunk 1 above)
Comment (lluunn) [reply docs_dev/tf_serving.md]: done

File: docs_dev/tf_serving.md (hunk 3 above)
Comment (lluunn) [reply docs_dev/tf_serving.md]: I think it's easier to make progress if we do it in our http proxy comparing to TF serving.

We can discuss on this thread https://github.com/kubeflow/kubeflow/issues/1000

Comment (jlewi) [reply docs_dev/tf_serving.md]: Can you add a link to the issue in the doc?

Comment (lluunn) [reply docs_dev/tf_serving.md]: It's there

**Comment by lluunn** [comment]
@jlewi PTAL, thanks!

**Comment by lluunn** [comment]
Error: No module named py.

**Comment by lluunn** [comment]
https://github.com/kubeflow/kubeflow/issues/1218

**Pull request merged**